app/views/admin/articles/index.html.erb

```diff
@@ -1,4 +1,13 @@
-<main class="row">
+<main
+  class="row"
+  data-controller="article-pinned-modal"
+  data-article-pinned-modal-root-selector-value="#article-pin-modal-root"
+  data-article-pinned-modal-content-selector-value="#article-pin-modal"
+  data-article-pinned-modal-title-value="There's another article pinned..."
+  data-article-pinned-modal-size-value="m"
+  data-article-pinned-modal-cancel-button-id-value="article-pin-modal-cancel"
+  data-article-pinned-modal-ok-button-id-value="article-pin-modal-ok"
+  data-action="article-pinned-modal:open@document->article-pinned-modal#openModal">
```

Comment:
where the magic happens. This tells Stimulus to listen to the custom event `article-pinned-modal:open` on the `document` object and send it to `ArticlePinnedModalController#openModal`. 

Remember this is dispatched in `ArticleController#pinArticle` after the checks the checkbox and the right conditions are met

Reply:
This is quite neat!